agent_test.go

```diff
@@ -39,3 +41,38 @@ func TestMain(m *testing.M) {
 	os.Exit(ret)
 
 }
+
+func TestParseKernelCmdline(t *testing.T) {
+	type args struct {
+		cmdline string
+	}
+
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{"Empty cmdline",
+			args{""}, false},
+		{"Log config info",
+			args{"agent.debug=info"}, false},
+		{"Log config debug",
+			args{"agent.debug=debug "}, false},
+		{"Log config warn",
+			args{" systemd kernel=someting agent.debug=warn "}, false},
+		{"Log config empty with space",
+			args{"agent.debug= debug "}, false},
+		{"Log config empty",
+			args{"agent.debug="}, false},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := parseKernelCmdline(tt.args.cmdline); (err != nil) != tt.wantErr {
+				t.Errorf("parseKernelCmdline() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			agentLog.Infof("info message")
+			agentLog.Debugf("debug message")
```

Comment:
Indeed, it does not help, I added to help me know if was working but agree I can add the Pipe sugguestion